src/random.h

```diff
@@ -89,6 +89,14 @@ void GetStrongRandBytes(unsigned char* buf, int num) noexcept;
  */
 void RandAddPeriodic();
 
+/**
+ * Gathers entropy from the low bits of the time at which events occur. Should
+ * be called with a few bytes describing the event at the time an event occurs.
+ *
+ * Thread-safe.
+ */
+void SeedEvent(const unsigned char* event_type_buf, size_t buf_len);
```

Comment:
Do we expect the argument to always be a string? The interface could be made a bit simpler and safer if that's the case (you could also avoid the size_t prefix, and instead just write the c_str() including null byte, which self-describes the length).

Reply:
I presume not, both for upcoming p2p encryption where the message type may just be a char, and I currently have some radio stuff hooked up that dumps RSSI information into this.

Reply:
I kind of prefer this general (ptr,size) interface to using C strings, whose use we're generally trying to avoid in bitcoin code. `std::slice` would be a safer interface, if only C++11 had it.

Reply:
Here is a more C++ish interface, that serializes (through the serialization framework) all arguments passed to SeedEvent (without any allocations): https://github.com/sipa/bitcoin/tree/bluematt_2019-11-rng-netmsg

Reply:
Oof, that's....quite a bigger diff. Are you worried about the current interface introducing bugs? I don't think it's particularly likely (nor are we likely to add calls to this *everywhere*), and it matches the rest of the random. interface.

Reply:
If you switch to SHA256 (which you did) it's much smaller.

It also enables efficiently adding extra information without first copying everything into a buffer (say, you could call `SeedEvent(msg.command, peerid);` just as easily)